internal/page/certificateprovider/confirm_dont_want_to_be_certificate_provider_logged_out.go

```diff
@@ -0,0 +1,74 @@
+package certificateprovider
+
+import (
+	"net/http"
+	"net/url"
+
+	"github.com/ministryofjustice/opg-go-common/template"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/actor"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/actor/actoruid"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/lpastore"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/page"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/validation"
+)
+
+type confirmDontWantToBeCertificateProviderDataLoggedOut struct {
+	App    page.AppData
+	Errors validation.List
+	Lpa    *lpastore.Lpa
+}
+
+func ConfirmDontWantToBeCertificateProviderLoggedOut(tmpl template.Template, shareCodeStore ShareCodeStore, lpaStoreResolvingService LpaStoreResolvingService, lpaStoreClient LpaStoreClient, donorStore DonorStore, sessionStore SessionStore) page.Handler {
+	return func(appData page.AppData, w http.ResponseWriter, r *http.Request) error {
+		session, err := sessionStore.LpaData(r)
+		if err != nil {
+			return err
+		}
+
+		ctx := page.ContextWithSessionData(r.Context(), &page.SessionData{LpaID: session.LpaID})
+
+		lpa, err := lpaStoreResolvingService.Get(ctx)
+		if err != nil {
+			return err
+		}
+
+		data := &confirmDontWantToBeCertificateProviderDataLoggedOut{
+			App: appData,
+			Lpa: lpa,
+		}
+
+		if r.Method == http.MethodPost {
+			if !lpa.SignedAt.IsZero() {
+				if err := lpaStoreClient.SendCertificateProviderOptOut(ctx, lpa.LpaUID, actoruid.Service); err != nil {
+					return err
+				}
+			} else {
+				donor, err := donorStore.GetAny(ctx)
+				if err != nil {
+					return err
+				}
+
+				donor.CertificateProvider = actor.CertificateProvider{}
+				donor.Tasks.CertificateProvider = actor.TaskNotStarted
+				donor.Tasks.CheckYourLpa = actor.TaskNotStarted
+
+				if err = donorStore.Put(ctx, donor); err != nil {
+					return err
+				}
+			}
+
+			shareCode, err := shareCodeStore.Get(r.Context(), actor.TypeCertificateProvider, r.URL.Query().Get("referenceNumber"))
+			if err != nil {
+				return err
+			}
+
+			if err := shareCodeStore.Delete(r.Context(), shareCode); err != nil {
```

Comment:
I think the share code should be deleted in the "enter opt out reference number" page, like we do with other share codes

Reply:
I'd originally thought about deleting straight away but if the CP decides not to go ahead they can't get back to this flow later or accept the invite if they continue in the role.

Reply:
that makes sense but it makes me hate having people do things when logged out a little bit more